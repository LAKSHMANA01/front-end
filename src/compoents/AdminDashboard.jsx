
import Navbar from "./Navbar";
import Dashbord from "./Dashbord";


function AdminDashboard() {
  return (
    <>
        {/* <Route path="/" element={<Home></Home>}/> */}
        <div className="flex">
          <div className="grow ml-16 md:ml-64 h-full lg:h-screen bg-gray-100 text-gray-900 dark:bg-gray-900 dark:text-white ">
            <Navbar />
            <Dashbord />
          </div>
        </div>
    </>
  );
}
export default AdminDashboard;

