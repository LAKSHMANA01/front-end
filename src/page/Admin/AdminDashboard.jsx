import React from "react";
import Sidebar from "../../compoents/Sidebar";
import Navbar from "../../compoents/Navbar";
import Dashbord from "../../compoents/Dashbord";
import { Outlet } from "react-router-dom"; // For rendering child routes
import ThemeContextProvider from "../../ContextAPI/ContextAPI";

function AdminDashboard() {
  return (
    <ThemeContextProvider>
      <div className="flex">
        {/* Sidebar is always visible */}
        <Sidebar />
        
        {/* Conditional rendering: Check if the route is "/admin/Tickets" */}
        <div className="grow ml-16 md:ml-64 h-full lg:h-screen bg-gray-100 text-gray-900 dark:bg-gray-900 dark:text-white">
          {/* Render Navbar and Dashbord for all routes except "/admin/Tickets" */}
        
          
          {/* Render the content based on the current route */}
          {/*  // This renders the content from the child route (like Tickets) */}
          <Outlet />
        </div>
      </div>
    </ThemeContextProvider>
  );
}

export default AdminDashboard;
