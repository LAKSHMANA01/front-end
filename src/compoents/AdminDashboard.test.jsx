// AdminDashboard.test.jsx
import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import AdminDashboard from './AdminDashboard';

// Mock the child components
jest.mock('./Navbar', () => () => <div data-testid="navbar">Navbar</div>);
jest.mock('./Dashbord', () => () => <div data-testid="dashbord">Dashbord</div>);

describe('AdminDashboard Component', () => {
  test('renders Sidebar, Navbar, and Dashbord components', () => {
    render(<AdminDashboard />);
    
    
    expect(screen.getByTestId('navbar')).toBeInTheDocument();
    expect(screen.getByTestId('navbar')).toHaveTextContent('Navbar');
    
    expect(screen.getByTestId('dashbord')).toBeInTheDocument();
    expect(screen.getByTestId('dashbord')).toHaveTextContent('Dashbord');
  });

  test('matches the snapshot', () => {
    const { container } = render(<AdminDashboard />);
    expect(container).toMatchSnapshot();
  });
});