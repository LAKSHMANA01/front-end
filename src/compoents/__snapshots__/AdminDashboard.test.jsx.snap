// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminDashboard Component matches the snapshot 1`] = `
<div>
  <div
    class="flex"
  >
    <div
      class="grow ml-16 md:ml-64 h-full lg:h-screen bg-gray-100 text-gray-900 dark:bg-gray-900 dark:text-white "
    >
      <div
        data-testid="navbar"
      >
        Navbar
      </div>
      <div
        data-testid="dashbord"
      >
        Dashbord
      </div>
    </div>
  </div>
</div>
`;
